src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs

```diff
@@ -78,7 +78,7 @@ internal abstract class CrossTargetSubscriptionHostBase : OnceInitializedOnceDis
 
         protected async Task AddInitialSubscriptionsAsync()
         {
-            using (_tasksService.LoadedProject())
```

Comment:
No `_tasksService.LoadedProject()` blocked until it got a lock, releasing the lock when it was disposed.  `_tasksService.LoadedProjectAsync` asynchronously waits for a lock, executes the lambda, and then released the lock.

---
In reply to: [121492671](https://github.com/dotnet/project-system/pull/2419#discussion_r121492671) [](ancestors = 121492671)